updated the installation page with tab based navigation

Musharraffaijaz opened the pull request:
**Kindly check the changes, all the installation steps are changed into tab-based navigation**
The PR resolves the Issue: [DOC: update installation page #811](https://github.com/numpy/numpy.org/issues/811)

![image](https://github.com/user-attachments/assets/ac6142d0-80dd-4598-835c-3183531a3fbc)

Musharraffaijaz commented:
The PR resolves the Issue: [DOC: update installation page #811](https://github.com/numpy/numpy.org/issues/811)

lucascolley reviewed (changes requested):
I think the widget in its current state leaves too little space for text on smaller screens, which is a problem on mobile:

![image](https://github.com/user-attachments/assets/a5752690-e87e-442a-8741-76d4697f7723)

Musharraffaijaz commented:
Alright, will do the changes.

Musharraffaijaz commented:
The latest commits are done using the theme's in build shortcode for tabs. Kindly check the Screen Shot below
![image](https://github.com/user-attachments/assets/2c87ed13-5fa1-4441-8f04-64b675e9e724)

Musharraffaijaz commented:
@jarrodmillman is it possible if we could change the styling of the tabs a bit maybe? Instead of having a border at the top can we just change it to bottom border only and a highlighting on hover?

lucascolley reviewed (commented):
I really like how this looks now! One thing - please could you keep the type stubs section separate (i.e. not as a tab)? It's basically something extra to look at after already understanding one of the tabs, rather than instead of the tabs.

Musharraffaijaz commented:
Great to know you liked it, sure will keep the type stubs separate.

jarrodmillman commented:
> @jarrodmillman is it possible if we could change the styling of the tabs a bit maybe? Instead of having a border at the top can we just change it to bottom border only and a highlighting on hover?

We are using the pydata-sphinx-theme styling for the scientific-python-hugo-theme. So you will need to get the changes to the tab styling implemented here: https://github.com/pydata/pydata-sphinx-theme/

mattip commented on content/en/install.md:
````diff
@@ -117,23 +115,30 @@ tool [`pixi`] are very similar to the steps for `uv`:
+pixi run python
+```
 
 See next steps in [the SciPy user guide][scipy-user-guide].
````
This should move down, it is applicable to all the tabs

Musharraffaijaz replied:
@lucascolley here

mattip reviewed (commented):
I really like this layout, in my opinion it is better UX

lucascolley reviewed (changes requested):
the hyperlinks on lines 17 to 20 don't work any more - I think they can just be removed now

Musharraffaijaz commented:
The latest changes includes:
Moved the _Type stubes_ from the tab based navigation
Removed the _hyper links_ from line 17 - 20
Moved the "_Try Scipy_" out of the tab based navigation
![image](https://github.com/user-attachments/assets/e3f49199-7f03-4836-a288-396c336b3d89)

Musharraffaijaz commented:
@lucascolley is there anything else which needs to be done here?

lucascolley commented on content/en/install.md:
````diff
@@ -216,10 +227,19 @@ For details on how to build from source, see
+{{</ tabs >}}
 
-## Installing with type stubs
+### Try out SciPy!
 
+```bash
+pixi run python
+```
````
These lines should be under the `pixi` section

Musharraffaijaz replied:
Thank you for your review, but @mattip above suggested to move this out as this is applicable to all tabs.

lucascolley replied:
I believe @mattip meant the line `See next steps in [the SciPy user guide][scipy-user-guide].`. `pixi run python` is only relevant if you have set up a `pixi` project!

Musharraffaijaz replied:
Alright

Musharraffaijaz replied:
@lucascolley kindly review the newer commits
![image](https://github.com/user-attachments/assets/c115344d-81e3-4352-8114-a06bd0df2b18)

lucascolley commented on content/en/install.md:
```diff
@@ -14,10 +14,10 @@ The recommended method of installing SciPy depends on your preferred workflow.
-- [Environment-based (e.g. `pip`, `conda`)](#environment-based) (the traditional workflow)
-- [System package managers](#system-package-managers) (not recommended)
-- [Building from source](#building-from-source) (for debugging and development)
+- **Project-based (e.g. `uv`, `pixi`)** (recommended for new users)
+- **Environment-based (e.g. `pip`, `conda`)** (the traditional workflow)
+- **System package managers (not recommended)**
+- **Building from source (for debugging and development)**
```
```suggestion
- **Project-based** (e.g. `uv`, `pixi`) *(recommended for new users)*
- **Environment-based** (e.g. `pip`, `conda`) *(the traditional workflow)*
- **System package managers** *(not recommended)*
- **Building from source** *(for debugging and development)*
```

lucascolley commented on content/en/install.md:
```diff
@@ -29,52 +29,50 @@ Installing type stubs may be required for
 
-        uv init try-scipy
-        cd try-scipy
+1. Install uv, following the instructions in the uv documentation.
```
the hyperlink has been removed - it should be restored

Musharraffaijaz replied:
Done with the said changes

Musharraffaijaz replied:
@lucascolley done with the changes. Thank you

lucascolley reviewed (approved):
thanks @Musharraffaijaz, this looks good to me and is a nice improvement!

Musharraffaijaz commented:
> thanks @Musharraffaijaz, this looks good to me and is a nice improvement!

Thank you for the acknowledgement.

Musharraffaijaz commented:
> thanks @Musharraffaijaz, this looks good to me and is a nice improvement!

Thanks @lucascolley, I'm glad it was helpful. I'm always looking to learn and improve, so I'd be keen to discuss any other areas in the projects where I could contribute.

lucascolley commented:
alright, merging - thanks again @Musharraffaijaz !
